crates/rattler_installs_packages/src/wheel_builder/build_environment.rs

```diff
@@ -6,21 +6,59 @@ use crate::resolve::{resolve, PinnedPackage, ResolveOptions};
 use crate::types::Artifact;
 use crate::wheel_builder::{build_requirements, WheelBuildError, WheelBuilder};
 use fs_err as fs;
+use parking_lot::RwLock;
 use pep508_rs::{MarkerEnvironment, Requirement};
 use std::collections::{HashMap, HashSet};
 use std::ffi::OsString;
+use std::ops::DerefMut;
 use std::path::{Path, PathBuf};
 use std::process::{Command, Output};
 use std::str::FromStr;
 
+#[derive(Debug)]
+enum DeleteOrPersist {
+    /// Delete the temp dir when the BuildEnvironment is dropped
+    Delete(Option<tempfile::TempDir>),
+    /// Persist the temp dir
+    Persist(PathBuf),
+}
+
+impl DeleteOrPersist {
+    /// Persist the temp dir
+    fn persist(&mut self) {
+        if let Self::Delete(dir) = self {
+            let path = dir.take();
```

Comment:
Almost, we can do:

```
        if let Self::Delete(temp_dir) = self {
            if let Some(path) = temp_dir.take() {
                // This operation makes sure that the tempdir is not deleted
                // when the BuildEnvironment is dropped
                *self = Self::Persist(path.into_path());
            }
        }
```

we need the `.take` because we are taking ownership.

Reply:
But good point changed it!